File: be/src/main/java/nl/zzpmatcher/profile/business/ProfileBuilder.java
package nl.zzpmatcher.profile.business;

import nl.zzpmatcher.profile.business.Profile;
import nl.zzpmatcher.profile.business.Tag;
import nl.zzpmatcher.profile.business.UpdateProfileCommand;

import java.util.Arrays;
import java.util.List;
import java.util.stream.Collectors;

public class ProfileBuilder {

    public static Profile build(UpdateProfileCommand updateProfileCommand) {

        final Profile profile = new Profile();
        profile.setId(updateProfileCommand.getUsername());

        final List<Tag> tagList = Arrays.stream(updateProfileCommand.getTags()).map(tag -> Tag.of(profile, tag)).collect(Collectors.toList());

        profile.setTags(tagList);

        return profile;
    }
}

File: be/src/main/java/nl/zzpmatcher/profile/business/ProfileBuilder.kt
package nl.zzpmatcher.profile.business

import java.util.*
import kotlin.streams.toList

object ProfileBuilder {

    fun build(updateProfileCommand: UpdateProfileCommand): Profile {

        val profile = Profile(id = updateProfileCommand.username)

        val tagList = updateProfileCommand.tags?.map { tag -> Tag(id = tag, profile = profile) }?.toList()

        profile.tags = tagList

        return profile
    }
}
